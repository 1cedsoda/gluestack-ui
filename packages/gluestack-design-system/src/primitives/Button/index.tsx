import React from 'react';
import { Button as GlueStackButton } from '@gluestack/ui';

export const Button = ({ ...props }) => {
  return <GlueStackButton {...props} />;
};
