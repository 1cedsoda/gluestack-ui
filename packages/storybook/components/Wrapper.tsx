// import React from 'react';
import { GluestackUIProvider, Center, config } from '@gluestack/ui';
// import { StyledProvider } from 'dank-style';
// import { set, get } from '@gluestack/color-mode';
// import { useDarkMode } from 'storybook-dark-mode';
import React from 'react';

// window['setTheme'] = set;
// window['getTheme'] = get;
const Wrapper = ({ children }: any) => {
  // let value = useDarkMode();
  // set(value ? 'dark' : 'light');

  // console.log('config', config);

  return (
    <GluestackUIProvider config={config}>
      <Center>{children}</Center>
    </GluestackUIProvider>
  );
};

Wrapper.displayName = 'GluestackUIProvider';

export default Wrapper;
