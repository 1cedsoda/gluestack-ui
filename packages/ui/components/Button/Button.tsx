import React from 'react';
import { Button, Center, AddIcon } from '@gluestack/ui';

export const Example = ({
  variant,
  text,
  size,
  isLoading,
  leftIcon,
  rightIcon,
  ...props
}) => {
  return (
    <Center>
      <Button variant={variant} size={size} {...props}>
        {isLoading && <Button.Spinner sx={{ style: { mr: 8 } }} />}
        {leftIcon && <AddIcon sx={{ style: { mr: 8 } }} />}
        <Button.Text>{text}</Button.Text>
        {rightIcon && <AddIcon sx={{ style: { ml: 8 } }} />}
      </Button>
    </Center>
  );
};
