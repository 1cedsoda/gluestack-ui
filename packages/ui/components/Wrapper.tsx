// import React from 'react';
import { Provider, Center } from '@gluestack/ui';
import { set, get } from '@gluestack/color-mode';
import { useDarkMode } from 'storybook-dark-mode';
import React from 'react';

export default ({ children }: any) => {
  let value = useDarkMode();
  set(value ? 'dark' : 'light');
  return (
    <Provider>
      <Center>{children}</Center>
    </Provider>
  );
};
