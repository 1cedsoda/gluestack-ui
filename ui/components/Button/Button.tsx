import React from 'react';
import { Button } from '@gluestack/ui';

export const Example = ({ ...props }) => {
  return (
    <Button sx={{ style: { bg: '$red.500' } }} {...props}>
      <Button.Text>Hello</Button.Text>
    </Button>
  );
};
