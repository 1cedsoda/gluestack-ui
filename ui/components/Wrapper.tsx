// import React from 'react';
import { GluestackUIProvider, Center } from '@gluestack/ui';
import { StyledProvider } from '@gluestack/ui-styled';
import { set, get } from '@gluestack/color-mode';
import { useDarkMode } from 'storybook-dark-mode';
import React from 'react';
import { config } from '../src/styled-components/ui.config';

const Wrapper = ({ children }: any) => {
  let value = useDarkMode();
  set(value ? 'dark' : 'light');

  console.log('config', config);

  return (
    <StyledProvider config={config}>
      <GluestackUIProvider>
        <Center>{children}</Center>
      </GluestackUIProvider>
    </StyledProvider>
  );
};

Wrapper.displayName = 'Wrapper';

export default Wrapper;
