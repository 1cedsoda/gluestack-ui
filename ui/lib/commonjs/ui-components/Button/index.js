'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true,
});
exports.Button = void 0;
var _styledComponents = require('../../styled-components');
var _uiCreator = require('@gluestack/ui-creator');
const Button = (0, _uiCreator.createButton)({
  StyledButton: _styledComponents.StyledButton,
  StyledButtonText: _styledComponents.StyledButtonText,
});
exports.Button = Button;
//# sourceMappingURL=index.js.map
