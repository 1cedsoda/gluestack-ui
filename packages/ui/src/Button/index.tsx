import Button from './Button';
import ButtonText from './ButtonText';

const ButtonTemp = Button as any;
ButtonTemp.Text = ButtonText;

export { ButtonTemp as Button };
