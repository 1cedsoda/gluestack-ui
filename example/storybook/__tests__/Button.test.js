import React from 'react';
import renderer from 'react-test-renderer';
// import { storiesOf } from '@storybook/react';
// import { action } from '@storybook/addon-actions';
import { render } from '@testing-library/react-native';

import { ButtonStory } from '../src/components/Forms/Button/Button';
import Wrapper from '../src/components/Wrapper';

describe('Button component', () => {
  it('matches snapshot', () => {
    //@ts-ignore
    const tree = renderer
      .create(
        <Wrapper>
          <ButtonStory />
        </Wrapper>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Button disabled', () => {
    const tree = renderer
      .create(
        <Wrapper>
          <ButtonStory isDisabled />
        </Wrapper>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Button hovered', () => {
    const tree = renderer
      .create(
        <Wrapper>
          <ButtonStory isHovered />
        </Wrapper>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('', () => {
    const tree = renderer
      .create(
        <Wrapper>
          <ButtonStory isPressed />
        </Wrapper>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should press button', () => {
    const mock = jest.fn();
    const tree = renderer
      .create(
        <Wrapper>
          <ButtonStory onPress={mock} />
        </Wrapper>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
