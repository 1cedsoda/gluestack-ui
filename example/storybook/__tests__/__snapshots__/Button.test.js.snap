// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component action negative 1`] = `
<div
  className="css-view-175oi2r"
  data-style="896xuz-1q64x7k 1njbp1t-913h4q"
  dir={null}
>
  <div
    className="css-view-175oi2r"
    data-style="wj8veg-1hh0pc0 cktard-17xz9t9"
    dir={null}
  >
    <div
      className="css-view-175oi2r"
      data-style="896xuz-1q64x7k 1njbp1t-913h4q"
      dir={null}
    >
      <div
        className="css-view-175oi2r"
        data-style="wj8veg-1hh0pc0 cktard-17xz9t9"
        dir={null}
      >
        <div
          className="css-view-175oi2r"
          data-style="wj8veg-1hh0pc0 3wzsaj-913h4q"
          dir={null}
        >
          <div
            className="css-view-175oi2r r-cursor-1loqt21 r-touchAction-1otgn73"
            data-style="fhf2c8-hnqgw2 fhf2c8-xe265e fhf2c8-1qr77c8 fhf2c8-mjo3f8 3wzsaj-913h4q"
            dir={null}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex="0"
          >
            <div
              className="css-text-1rynq56"
              data-style="727xeq-10883ij fhf2c8-1eydhex fhf2c8-1vrz3gw fhf2c8-14oe7p6 fhf2c8-11hczkh 3wzsaj-qc4xxi 1njbp1t-qc4xxi 1njbp1t-qc4xxi 3wzsaj-913h4q"
              dir="auto"
            >
              Button
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Button component action positive 1`] = `
<div
  className="css-view-175oi2r"
  data-style="896xuz-1q64x7k 1njbp1t-913h4q"
  dir={null}
>
  <div
    className="css-view-175oi2r"
    data-style="wj8veg-1hh0pc0 cktard-17xz9t9"
    dir={null}
  >
    <div
      className="css-view-175oi2r"
      data-style="896xuz-1q64x7k 1njbp1t-913h4q"
      dir={null}
    >
      <div
        className="css-view-175oi2r"
        data-style="wj8veg-1hh0pc0 cktard-17xz9t9"
        dir={null}
      >
        <div
          className="css-view-175oi2r"
          data-style="wj8veg-1hh0pc0 3wzsaj-913h4q"
          dir={null}
        >
          <div
            className="css-view-175oi2r r-cursor-1loqt21 r-touchAction-1otgn73"
            data-style="fhf2c8-hnqgw2 fhf2c8-lxtrsj fhf2c8-1qr77c8 fhf2c8-mjo3f8 3wzsaj-913h4q"
            dir={null}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex="0"
          >
            <div
              className="css-text-1rynq56"
              data-style="727xeq-10883ij fhf2c8-1eydhex fhf2c8-3noqo9 fhf2c8-14oe7p6 fhf2c8-11hczkh 3wzsaj-qc4xxi 1njbp1t-qc4xxi 1njbp1t-qc4xxi 3wzsaj-913h4q"
              dir="auto"
            >
              Button
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Button component action primary 1`] = `
<div
  className="css-view-175oi2r"
  data-style="896xuz-1q64x7k 1njbp1t-913h4q"
  dir={null}
>
  <div
    className="css-view-175oi2r"
    data-style="wj8veg-1hh0pc0 cktard-17xz9t9"
    dir={null}
  >
    <div
      className="css-view-175oi2r"
      data-style="896xuz-1q64x7k 1njbp1t-913h4q"
      dir={null}
    >
      <div
        className="css-view-175oi2r"
        data-style="wj8veg-1hh0pc0 cktard-17xz9t9"
        dir={null}
      >
        <div
          className="css-view-175oi2r"
          data-style="wj8veg-1hh0pc0 3wzsaj-913h4q"
          dir={null}
        >
          <div
            className="css-view-175oi2r r-cursor-1loqt21 r-touchAction-1otgn73"
            data-style="fhf2c8-hnqgw2 fhf2c8-58zsu5 fhf2c8-1qr77c8 fhf2c8-mjo3f8 3wzsaj-913h4q"
            dir={null}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex="0"
          >
            <div
              className="css-text-1rynq56"
              data-style="727xeq-10883ij fhf2c8-1eydhex fhf2c8-7s6zs0 fhf2c8-14oe7p6 fhf2c8-11hczkh 3wzsaj-qc4xxi 1njbp1t-qc4xxi 1njbp1t-qc4xxi 3wzsaj-913h4q"
              dir="auto"
            >
              Button
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Button component action secondary 1`] = `
<div
  className="css-view-175oi2r"
  data-style="896xuz-1q64x7k 1njbp1t-913h4q"
  dir={null}
>
  <div
    className="css-view-175oi2r"
    data-style="wj8veg-1hh0pc0 cktard-17xz9t9"
    dir={null}
  >
    <div
      className="css-view-175oi2r"
      data-style="896xuz-1q64x7k 1njbp1t-913h4q"
      dir={null}
    >
      <div
        className="css-view-175oi2r"
        data-style="wj8veg-1hh0pc0 cktard-17xz9t9"
        dir={null}
      >
        <div
          className="css-view-175oi2r"
          data-style="wj8veg-1hh0pc0 3wzsaj-913h4q"
          dir={null}
        >
          <div
            className="css-view-175oi2r r-cursor-1loqt21 r-touchAction-1otgn73"
            data-style="fhf2c8-hnqgw2 fhf2c8-8r4mg3 fhf2c8-1qr77c8 fhf2c8-mjo3f8 3wzsaj-913h4q"
            dir={null}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex="0"
          >
            <div
              className="css-text-1rynq56"
              data-style="727xeq-10883ij fhf2c8-1eydhex fhf2c8-1wt7l86 fhf2c8-14oe7p6 fhf2c8-11hczkh 3wzsaj-qc4xxi 1njbp1t-qc4xxi 1njbp1t-qc4xxi 3wzsaj-913h4q"
              dir="auto"
            >
              Button
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Button component size lg 1`] = `
<div
  className="css-view-175oi2r"
  data-style="896xuz-1q64x7k 1njbp1t-913h4q"
  dir={null}
>
  <div
    className="css-view-175oi2r"
    data-style="wj8veg-1hh0pc0 cktard-17xz9t9"
    dir={null}
  >
    <div
      className="css-view-175oi2r"
      data-style="896xuz-1q64x7k 1njbp1t-913h4q"
      dir={null}
    >
      <div
        className="css-view-175oi2r"
        data-style="wj8veg-1hh0pc0 cktard-17xz9t9"
        dir={null}
      >
        <div
          className="css-view-175oi2r"
          data-style="wj8veg-1hh0pc0 3wzsaj-913h4q"
          dir={null}
        >
          <div
            className="css-view-175oi2r r-cursor-1loqt21 r-touchAction-1otgn73"
            data-style="fhf2c8-hnqgw2 fhf2c8-58zsu5 fhf2c8-1qr77c8 fhf2c8-1m88kvk 3wzsaj-913h4q"
            dir={null}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex="0"
          >
            <div
              className="css-text-1rynq56"
              data-style="727xeq-10883ij fhf2c8-1eydhex fhf2c8-7s6zs0 fhf2c8-14oe7p6 fhf2c8-1n3pgep 3wzsaj-qc4xxi 1njbp1t-qc4xxi 1njbp1t-qc4xxi 3wzsaj-913h4q"
              dir="auto"
            >
              Button
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Button component size md 1`] = `
<div
  className="css-view-175oi2r"
  data-style="896xuz-1q64x7k 1njbp1t-913h4q"
  dir={null}
>
  <div
    className="css-view-175oi2r"
    data-style="wj8veg-1hh0pc0 cktard-17xz9t9"
    dir={null}
  >
    <div
      className="css-view-175oi2r"
      data-style="896xuz-1q64x7k 1njbp1t-913h4q"
      dir={null}
    >
      <div
        className="css-view-175oi2r"
        data-style="wj8veg-1hh0pc0 cktard-17xz9t9"
        dir={null}
      >
        <div
          className="css-view-175oi2r"
          data-style="wj8veg-1hh0pc0 3wzsaj-913h4q"
          dir={null}
        >
          <div
            className="css-view-175oi2r r-cursor-1loqt21 r-touchAction-1otgn73"
            data-style="fhf2c8-hnqgw2 fhf2c8-58zsu5 fhf2c8-1qr77c8 fhf2c8-mjo3f8 3wzsaj-913h4q"
            dir={null}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex="0"
          >
            <div
              className="css-text-1rynq56"
              data-style="727xeq-10883ij fhf2c8-1eydhex fhf2c8-7s6zs0 fhf2c8-14oe7p6 fhf2c8-11hczkh 3wzsaj-qc4xxi 1njbp1t-qc4xxi 1njbp1t-qc4xxi 3wzsaj-913h4q"
              dir="auto"
            >
              Button
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Button component size sm 1`] = `
<div
  className="css-view-175oi2r"
  data-style="896xuz-1q64x7k 1njbp1t-913h4q"
  dir={null}
>
  <div
    className="css-view-175oi2r"
    data-style="wj8veg-1hh0pc0 cktard-17xz9t9"
    dir={null}
  >
    <div
      className="css-view-175oi2r"
      data-style="896xuz-1q64x7k 1njbp1t-913h4q"
      dir={null}
    >
      <div
        className="css-view-175oi2r"
        data-style="wj8veg-1hh0pc0 cktard-17xz9t9"
        dir={null}
      >
        <div
          className="css-view-175oi2r"
          data-style="wj8veg-1hh0pc0 3wzsaj-913h4q"
          dir={null}
        >
          <div
            className="css-view-175oi2r r-cursor-1loqt21 r-touchAction-1otgn73"
            data-style="fhf2c8-hnqgw2 fhf2c8-58zsu5 fhf2c8-1qr77c8 fhf2c8-npzoh9 3wzsaj-913h4q"
            dir={null}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex="0"
          >
            <div
              className="css-text-1rynq56"
              data-style="727xeq-10883ij fhf2c8-1eydhex fhf2c8-7s6zs0 fhf2c8-14oe7p6 fhf2c8-1q2dyj6 3wzsaj-qc4xxi 1njbp1t-qc4xxi 1njbp1t-qc4xxi 3wzsaj-913h4q"
              dir="auto"
            >
              Button
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Button component size xl 1`] = `
<div
  className="css-view-175oi2r"
  data-style="896xuz-1q64x7k 1njbp1t-913h4q"
  dir={null}
>
  <div
    className="css-view-175oi2r"
    data-style="wj8veg-1hh0pc0 cktard-17xz9t9"
    dir={null}
  >
    <div
      className="css-view-175oi2r"
      data-style="896xuz-1q64x7k 1njbp1t-913h4q"
      dir={null}
    >
      <div
        className="css-view-175oi2r"
        data-style="wj8veg-1hh0pc0 cktard-17xz9t9"
        dir={null}
      >
        <div
          className="css-view-175oi2r"
          data-style="wj8veg-1hh0pc0 3wzsaj-913h4q"
          dir={null}
        >
          <div
            className="css-view-175oi2r r-cursor-1loqt21 r-touchAction-1otgn73"
            data-style="fhf2c8-hnqgw2 fhf2c8-58zsu5 fhf2c8-1qr77c8 3wzsaj-913h4q"
            dir={null}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex="0"
          >
            <div
              className="css-text-1rynq56"
              data-style="727xeq-10883ij fhf2c8-1eydhex fhf2c8-7s6zs0 fhf2c8-14oe7p6 3wzsaj-qc4xxi 1njbp1t-qc4xxi 1njbp1t-qc4xxi 3wzsaj-913h4q"
              dir="auto"
            >
              Button
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Button component size xs 1`] = `
<div
  className="css-view-175oi2r"
  data-style="896xuz-1q64x7k 1njbp1t-913h4q"
  dir={null}
>
  <div
    className="css-view-175oi2r"
    data-style="wj8veg-1hh0pc0 cktard-17xz9t9"
    dir={null}
  >
    <div
      className="css-view-175oi2r"
      data-style="896xuz-1q64x7k 1njbp1t-913h4q"
      dir={null}
    >
      <div
        className="css-view-175oi2r"
        data-style="wj8veg-1hh0pc0 cktard-17xz9t9"
        dir={null}
      >
        <div
          className="css-view-175oi2r"
          data-style="wj8veg-1hh0pc0 3wzsaj-913h4q"
          dir={null}
        >
          <div
            className="css-view-175oi2r r-cursor-1loqt21 r-touchAction-1otgn73"
            data-style="fhf2c8-hnqgw2 fhf2c8-58zsu5 fhf2c8-1qr77c8 fhf2c8-1yhbyhf 3wzsaj-913h4q"
            dir={null}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex="0"
          >
            <div
              className="css-text-1rynq56"
              data-style="727xeq-10883ij fhf2c8-1eydhex fhf2c8-7s6zs0 fhf2c8-14oe7p6 fhf2c8-193yier 3wzsaj-qc4xxi 1njbp1t-qc4xxi 1njbp1t-qc4xxi 3wzsaj-913h4q"
              dir="auto"
            >
              Button
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Button component variant link 1`] = `
<div
  className="css-view-175oi2r"
  data-style="896xuz-1q64x7k 1njbp1t-913h4q"
  dir={null}
>
  <div
    className="css-view-175oi2r"
    data-style="wj8veg-1hh0pc0 cktard-17xz9t9"
    dir={null}
  >
    <div
      className="css-view-175oi2r"
      data-style="896xuz-1q64x7k 1njbp1t-913h4q"
      dir={null}
    >
      <div
        className="css-view-175oi2r"
        data-style="wj8veg-1hh0pc0 cktard-17xz9t9"
        dir={null}
      >
        <div
          className="css-view-175oi2r"
          data-style="wj8veg-1hh0pc0 3wzsaj-913h4q"
          dir={null}
        >
          <div
            className="css-view-175oi2r r-cursor-1loqt21 r-touchAction-1otgn73"
            data-style="fhf2c8-hnqgw2 fhf2c8-58zsu5 fhf2c8-1qr77c8 fhf2c8-mjo3f8 3wzsaj-913h4q"
            dir={null}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex="0"
          >
            <div
              className="css-text-1rynq56"
              data-style="727xeq-10883ij fhf2c8-1eydhex fhf2c8-7s6zs0 fhf2c8-14oe7p6 fhf2c8-11hczkh 3wzsaj-qc4xxi 1njbp1t-qc4xxi 1njbp1t-qc4xxi 3wzsaj-913h4q"
              dir="auto"
            >
              Button
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Button component variant outline 1`] = `
<div
  className="css-view-175oi2r"
  data-style="896xuz-1q64x7k 1njbp1t-913h4q"
  dir={null}
>
  <div
    className="css-view-175oi2r"
    data-style="wj8veg-1hh0pc0 cktard-17xz9t9"
    dir={null}
  >
    <div
      className="css-view-175oi2r"
      data-style="896xuz-1q64x7k 1njbp1t-913h4q"
      dir={null}
    >
      <div
        className="css-view-175oi2r"
        data-style="wj8veg-1hh0pc0 cktard-17xz9t9"
        dir={null}
      >
        <div
          className="css-view-175oi2r"
          data-style="wj8veg-1hh0pc0 3wzsaj-913h4q"
          dir={null}
        >
          <div
            className="css-view-175oi2r r-cursor-1loqt21 r-touchAction-1otgn73"
            data-style="fhf2c8-hnqgw2 fhf2c8-58zsu5 fhf2c8-1qr77c8 fhf2c8-mjo3f8 3wzsaj-913h4q"
            dir={null}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex="0"
          >
            <div
              className="css-text-1rynq56"
              data-style="727xeq-10883ij fhf2c8-1eydhex fhf2c8-7s6zs0 fhf2c8-14oe7p6 fhf2c8-11hczkh 3wzsaj-qc4xxi 1njbp1t-qc4xxi 1njbp1t-qc4xxi 3wzsaj-913h4q"
              dir="auto"
            >
              Button
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Button component variant solid 1`] = `
<div
  className="css-view-175oi2r"
  data-style="896xuz-1q64x7k 1njbp1t-913h4q"
  dir={null}
>
  <div
    className="css-view-175oi2r"
    data-style="wj8veg-1hh0pc0 cktard-17xz9t9"
    dir={null}
  >
    <div
      className="css-view-175oi2r"
      data-style="896xuz-1q64x7k 1njbp1t-913h4q"
      dir={null}
    >
      <div
        className="css-view-175oi2r"
        data-style="wj8veg-1hh0pc0 cktard-17xz9t9"
        dir={null}
      >
        <div
          className="css-view-175oi2r"
          data-style="wj8veg-1hh0pc0 3wzsaj-913h4q"
          dir={null}
        >
          <div
            className="css-view-175oi2r r-cursor-1loqt21 r-touchAction-1otgn73"
            data-style="fhf2c8-hnqgw2 fhf2c8-58zsu5 fhf2c8-1qr77c8 fhf2c8-mjo3f8 3wzsaj-913h4q"
            dir={null}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex="0"
          >
            <div
              className="css-text-1rynq56"
              data-style="727xeq-10883ij fhf2c8-1eydhex fhf2c8-7s6zs0 fhf2c8-14oe7p6 fhf2c8-11hczkh 3wzsaj-qc4xxi 1njbp1t-qc4xxi 1njbp1t-qc4xxi 3wzsaj-913h4q"
              dir="auto"
            >
              Button
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
